### PR TITLE
Principal Serial Number Radix feature

### LINKED ISSUE #2469: X509SerialNumberPrincipalResolver to support HEX numbers
I have included the following in my report:

- [] CAS server version 5.0.2
- [] Container/environment any

For smooth integration with different other systems it would be nice to support alternative number formats.

### PR DESCRIPTION
**Pull request by fedon**
Closes #2469 

Ensure that you include the following:

- [] X509SerialNumberPrincipalResolver extended to support HEX and other configurable radixes.
- [] cas.authn.x509.principal.radix=16
cas.authn.x509.principal.hexZeroPadding=true
- [] Character.MIN_RADIX <= radix <= Character.MAX_RADIX
- [] N/A.

### CODE REVIEW

**Comment by mmoayyed** [comment]
Excellent work. Comments follow.

File: core/cas-server-core-configuration/src/main/java/org/apereo/cas/configuration/model/core/authentication/PersonDirPrincipalResolverProperties.java [hunk 1]
```diff
@@ -13,6 +13,9 @@
     private boolean returnNull;
     private boolean principalResolutionFailureFatal;
 
+    private int radix;
```
Comment (mmoayyed) [inline core/cas-server-core-configuration/src/main/java/org/apereo/cas/configuration/model/core/authentication/PersonDirPrincipalResolverProperties.java]: I don't think these belong here. You need to move them down to the x509 level.

**Comment by mmoayyed** [comment]
Excellent work. Could you also update the docs to advertise the new settings now available?

**Comment by fedon** [comment]
Sure, I'll update settings docs.

**Comment by mmoayyed** [comment]
Wonderful. Thanks very much. With build passing, I think we should be good to merge.

**Comment by fedon** [comment]
Travis failed on support:cas-server-support-validation:test which I didn't touch. It was docs commit :)

**Pull request merged**